docs/cody/capabilities/chat.mdx

```diff
@@ -103,5 +105,93 @@ You are provided with code suggestions in the chat window and the following opti
+- A supported editor extension (VS Code, JetBrains) installed or use via Web app
+
+## Key features
+
+The enhanced chat experience includes everything in the Free plan, plus the following:
+
+## Intent detection
+
+Intent detection automatically analyzes user queries and determines whether to provide an AI chat or search response. This functionality helps simplify developer workflows by providing the most appropriate type of response without requiring explicit mode switching.
+
+### How it works
+
+When a user submits a query in the chat panel, the intent detection component:
+
+- Analyzes the query content and structure
+- Determines the most appropriate response type (search or chat)
+- Returns results in the optimal format
+- Provides the ability to toggle between response types manually
+
+Let's look at an example of how this might work:
+
+#### Search-based response
+
+![Intent detection code search response](https://storage.googleapis.com/sourcegraph-assets/Docs/intent-detection-code-search-response-01242025.jpg)
+
+#### Chat-based response
+
+![Intent detection chat response](https://storage.googleapis.com/sourcegraph-assets/Docs/intent-detection-chat-response-01242025.jpg)
+
+## Smart search integration
+
+The smart search integration enhances Sourcegraph's chat experience by providing lightweight code search capabilities directly within the chat interface. This feature simplifies developer workflows by offering quick access to code search without leaving the chat environment.
+
+The integration delivers personalized search results ranked by your contribution history, with frequently accessed repositories appearing higher in results. Users can view code snippets with relevant context and open files directly in their editor.
+
+Search results automatically become available as context for follow-up queries, with flexible controls for selecting which results to include. While optimized for keyword-style queries and searching across a few repositories, this integration complements rather than replaces the full [Code Search](/code-search) product, which remains the recommended tool for comprehensive enterprise-wide code search.
+
+## Context-aware responses
+
+Search results generated through smart search integration can be automatically used as context for follow-up queries. Here's what happens with each scenario:
+
+### Search responses
+
+* Search results can be used directly as context for follow-up queries
+* Users can select which search results to include as context using checkboxes
+* By default, all search results are added as context
+* A context chip shows the number of search results being used (e.g., "10 code search results")
+* Users can remove the context chip if they don't want to use it for follow-ups
+
+### Chat responses
+
+* Performs background searches for context
+* Retrieves full context from files, symbols, remote repos, and web pages
+* Can execute terminal commands (with permission) for additional context
```

Comment:
do we want to include this since its still experimental? i think we should at least call out this is experimental

Reply:
Done!